src/SW_Site.c

```diff
@@ -1296,20 +1300,22 @@ void add_deepdrain_layer(void) {
 @brief First time called with no layers so SW_Site.lyr
  not initialized yet, malloc() required.  For each
  layer thereafter realloc() is called.
+
+@param[in,out] SW_Site Struct of type SW_SITE describing the simulated site
 */
-LyrIndex _newlayer(void) {
+LyrIndex _newlayer(SW_SITE* SW_Site) {
+	int n_layers = SW_Site->n_layers;
 
-	SW_SITE *v = &SW_Site;
-	v->n_layers++;
+	SW_Site->n_layers++;
 
-	v->lyr = (!v->lyr) /* if not yet defined */
-		? (SW_LAYER_INFO **) Mem_Calloc(v->n_layers, sizeof(SW_LAYER_INFO *), "_newlayer()") /* malloc() it  */
-		: (SW_LAYER_INFO **) Mem_ReAlloc(v->lyr, sizeof(SW_LAYER_INFO *) * (v->n_layers)); /* else realloc() */
+	SW_Site->lyr = (!SW_Site->lyr) /* if not yet defined */
+		? (SW_LAYER_INFO **) Mem_Calloc(SW_Site->n_layers, sizeof(SW_LAYER_INFO *), "_newlayer()") /* malloc() it  */
+		: (SW_LAYER_INFO **) Mem_ReAlloc(SW_Site->lyr, sizeof(SW_LAYER_INFO *) * (SW_Site->n_layers)); /* else realloc() */
 
-	v->lyr[v->n_layers - 1] = (SW_LAYER_INFO *) Mem_Calloc(1, sizeof(SW_LAYER_INFO), "_newlayer()");

```

Comment:
The variable "lyr" is still base0. At the beginning of the function, the value is stored in a variable called `n_layers` and then `SW_Site->layers` is incremented. So when the function needs the original or incremented value, it will use `SW_Site->layers` or `n_layers`, respectively. This was done to reduce the instances of decrementing by one in multiple spots.

Reply:
make sense, nice implementation! My bad for not realizing the smarter way that you implemented